src/resources/prefabs.rs

```diff
@@ -1,48 +1,96 @@
     ecs::World,
+    utils::application_root_dir,
 };
 
-use crate::components::creatures::{CreaturePrefabData, CreatureType};
+use crate::components::creatures::CreaturePrefabData;
 
 #[derive(Default)]
 pub struct CreaturePrefabs {
-    prefabs: HashMap<CreatureType, Handle<Prefab<CreaturePrefabData>>>,
+    prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>,
 }
 
 impl CreaturePrefabs {
     pub fn insert(
         &mut self,
-        creature_type: CreatureType,
+        creature_type: String,
         prefab_handle: Handle<Prefab<CreaturePrefabData>>,
     ) {
         self.prefabs.insert(creature_type, prefab_handle);
     }
 
-    pub fn get_prefab(
-        &self,
-        creature_type: &CreatureType,
-    ) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
+    pub fn get_prefab(&self, creature_type: &str) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
         self.prefabs.get(creature_type)
     }
+
+    pub fn get_prefabs(&self) -> &HashMap<String, Handle<Prefab<CreaturePrefabData>>> {
+        &self.prefabs
+    }
+
+    pub fn set_prefabs(&mut self, prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>) {
+        self.prefabs = prefabs;
+    }
 }
 
 // Here we load all prefabs for the different creatures in the game.
 // These prefabs are then stored in a resource of type CreaturePrefabs that is used by the spawner system.
-pub fn initialize_prefabs(world: &mut World) {
+pub fn initialize_prefabs(world: &mut World) -> ProgressCounter {
     let mut creature_prefabs = CreaturePrefabs::default();
-    let (carnivore_prefab, herbivore_prefab, plant_prefab) =
-        world.exec(|loader: PrefabLoader<'_, CreaturePrefabData>| {
-            (
-                loader.load("prefabs/carnivore.ron", RonFormat, (), ()),
-                loader.load("prefabs/herbivore.ron", RonFormat, (), ()),
-                loader.load("prefabs/plant.ron", RonFormat, (), ()),
-            )
-        });
-    creature_prefabs.insert("Carnivore".to_string(), carnivore_prefab);
-    creature_prefabs.insert("Herbivore".to_string(), herbivore_prefab);
-    creature_prefabs.insert("Plant".to_string(), plant_prefab);
+    let mut progress_counter = ProgressCounter::new();
+    let prefab_iter = {
+        let prefab_dir_path = application_root_dir() + "/resources/prefabs";
+        let prefab_iter = read_dir(prefab_dir_path).unwrap();
+        prefab_iter.map(|prefab_dir_entry| {
+            world.exec(|loader: PrefabLoader<'_, CreaturePrefabData>| {
+                let prefab_path_buf = prefab_dir_entry.unwrap().path();
+                let prefab_filename = prefab_path_buf.file_name().unwrap();
+                loader.load(
+                    "prefabs/".to_string() + prefab_filename.to_str().unwrap(),
+                    RonFormat,
+                    (),
+                    &mut progress_counter,
+                )
+            })
+        })
+    };
+
+    let mut count = 0;
+    for prefab in prefab_iter {
```

Comment:
you could use `for (index, prefab) in prefab_iter.enumerate()` here I think, see https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate